zerver/lib/push_notifications.py

```diff
@@ -314,6 +315,17 @@ def get_apns_payload(message):
         'message_ids': [message.id],
     }
 
+def relative_to_full_url(user_profile, content):
+    # type: (UserProfile, Text) -> Text
+
+    # URLs for uploaded avatars are of the form
+    # "/user_avatars/*".
+    # Make them full paths containing the realm URL.
+
+    return re.sub(
+        r"/user_avatars/(\S*)",
+        user_profile.realm.uri.rstrip("/") + r"/user_avatars/\1", content)
```

Comment:
`urllib.parse.urljoin` is the correct way to do this combination stuff in Python.